charts/kargo/templates/api/deployment.yaml

```diff
@@ -97,20 +97,46 @@ spec:
 {{- end }}
             initialDelaySeconds: 5
 {{- end }}
-{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
+{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled .Values.api.cabundle.enabled }}
           volumeMounts:
+          {{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
             - mountPath: /etc/kargo
               name: config
               readOnly: true
+          {{- end }}
+          {{- if .Values.api.cabundle.enabled }}
+            - mountPath: /etc/ssl/certs
+              name: certs
+          {{- end }}
 {{- end }}
 {{- with .Values.api.securityContext | default .Values.global.securityContext }}
           securityContext:
             {{- toYaml . | nindent 12 }}
 {{- end }}
           resources:
             {{- toYaml .Values.api.resources | nindent 12 }}
-{{- if or .Values.kubeconfigSecrets.kargo (and .Values.api.oidc.enabled .Values.api.oidc.dex.enabled) .Values.api.tls.enabled }}
+      {{- if .Values.api.cabundle.enabled }}
+      initContainers:
+      - name: parse-cabundle
+        image: docker.io/alpine/curl:latest
+        command:
+        - "/bin/sh"
+        - "-c"
+        args:
+        - |
+          for FILE in /tmp/source/*; do awk 'BEGIN {c=0;} /BEGIN CERT/{c++} { print > "/usr/local/share/ca-certificates/cert." c ".crt"}' < $FILE; done
+          /usr/sbin/update-ca-certificates
+          rm -rf /tmp/target/*
+          cp -r /etc/ssl/certs/* /tmp/target/
```

Comment:
If we are going to delete `/tmp/target/` and rebuild its contents from scratch every time we start up, what is the point of the volume?

Reply:
Also, what is the significance of `/tmp/target/`?

Other than the fact that we're deleting it and re-creating it here, I don't see any other references to it. What is reading certs from that location?

Reply:
Ok... I see now. The volume mounted to `/tmp/target/` in the init container is mounted as `/etc/ssl/certs/` in the main container.

Have you tested this process end-to-end? I believe I still see some problems:

1. When you run `update-ca-certificates` to update the contents of `/etc/ssl/certs/`, that directory will contain _symlinks_ back to the certs in `/usr/local/share/ca-certificates/`. Those symlinks are being copied to `/tmp/target/`, which is mounted to `/etc/ssl/certs/` in the main container. Won't `/etc/ssl/certs/` in the main container then contain symlinks that cannot be resolved?

1. While I _do_ understand the point of the emptydir volume, I cannot see the rationale for the PVC option, since, as previously noted, you are deleting and recreating everything on each initialization. _What are we trying to persist if it's going to be fully replaced each time?_

It seems to me that if we were to continue along the init container route (which I am still not convinced is ideal), we would do well to do the following:

1. Do not bother with the persistent volume option.

1. Use a more appropriate image for the init container. idk why we should use `alpine/curl` when `curl` is not even a tool required for this process.

1. Mount an emptydir volume to `/tmp/target/` in the init container and mount the same volume to `/etc/ssl/certs` in the main container. (Like you're already doing.) 

1. Mount an emptydir volume to `/usr/local/share/ca-certificates/` in both containers. This is to avoid the dangling symlinks.

1. Update the script:
    1. Fix the bug in the for loop
    1. Don't both to `rm -rf /tmp/target/*` because we'll always be dealing with an empty dir.

I think that would work, but I also anticipate we'll eventually refactor all of the above so the binary in the main container just deals with certs at startup. This would be to avoid contending with the CVEs of a second image on an ongoing basis. So idk if it amounts to wasted effort to no just _start_ with that approach, which is in many ways quite a bit simpler regardless.